ogre2/src/Ogre2InertiaVisual.cc

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "ignition/rendering/ogre2/Ogre2InertiaVisual.hh"
+#include "ignition/rendering/ogre2/Ogre2Material.hh"
+#include "ignition/rendering/ogre2/Ogre2DynamicRenderable.hh"
+
+#ifdef _MSC_VER
+  #pragma warning(push, 0)
+#endif
+#include <OgreSceneNode.h>
+#ifdef _MSC_VER
+  #pragma warning(pop)
+#endif
+
+using namespace ignition;
+using namespace rendering;
+
+class ignition::rendering::Ogre2InertiaVisualPrivate
+{
+  /// \brief inertia visual materal
+  public: Ogre2MaterialPtr material = nullptr;
+
+  /// \brief Ogre renderable used to render the cross lines.
+  public: std::shared_ptr<Ogre2DynamicRenderable> crossLines = nullptr;
+
+  /// \brief Box visual
+  public: VisualPtr boxVis = nullptr;
+};
+
+//////////////////////////////////////////////////
+Ogre2InertiaVisual::Ogre2InertiaVisual()
+  : dataPtr(new Ogre2InertiaVisualPrivate)
+{
+}
+
+//////////////////////////////////////////////////
+Ogre2InertiaVisual::~Ogre2InertiaVisual()
+{
+}
+
+//////////////////////////////////////////////////
+void Ogre2InertiaVisual::PreRender()
+{
+}
+
+//////////////////////////////////////////////////
+void Ogre2InertiaVisual::Init()
+{
+  BaseInertiaVisual::Init();
+}
+
+//////////////////////////////////////////////////
+void Ogre2InertiaVisual::Destroy()
+{
+  if (this->dataPtr->boxVis != nullptr)
+  {
+    this->dataPtr->boxVis->Destroy();
+  }
+  BaseInertiaVisual::Destroy();
```

Comment:
Destroy material:

```
  if (this->dataPtr->material && this->Scene())
  {
    this->Scene()->DestroyMaterial(this->dataPtr->material);
    this->dataPtr->material.reset();
  }
```

Reply:
I am getting this message in the console 
```
visualization_demo: /var/lib/jenkins/workspace/ogre-2.1-debbuilder/repo/OgreMain/src/OgreHlmsDatablock.cpp:170: virtual Ogre::HlmsDatablock::~HlmsDatablock(): Assertion `mLinkedRenderables.empty() && "This Datablock is still being used by some Renderables." " Change their Datablocks before destroying this."' failed.
Aborted (core dumped)
```

I think this is because of `crossLines` still using the material. Explicitly destroying `crossLines` before the material is giving an exception in the test
```
      Start 17: UNIT_InertiaVisual_TEST
17/93 Test #17: UNIT_InertiaVisual_TEST ...............Child aborted***Exception:   0.33 sec
```

Reply:
Fixed in [`dd42817`](https://github.com/ignitionrobotics/ign-rendering/pull/326/commits/dd42817dd2b396b738ff0988850e433f41f1cd4a)